packages/go_router/lib/src/logging.dart

```diff
@@ -4,22 +4,35 @@
 
 import 'dart:async';
 import 'dart:developer' as developer;
+
 import 'package:flutter/foundation.dart';
 import 'package:logging/logging.dart';
 
 /// The logger for this package.
-final Logger log = Logger('GoRouter');
+@visibleForTesting
+final Logger logger = Logger('GoRouter');
+
+/// Whether or not the logging is enabled.
+bool _enabled = false;
+
+/// Logs the message if logging is enabled.
+void log(String message) {
```

Comment:
use `log` may be ambiguous if we later add other level for some reason. 

should this just set logger.level = Level.OFF when set enabled?

Reply:
> use log may be ambiguous if we later add other level for some reason.

We could later on change the signature to

```dart
void log(String message, [Level level = Level.INFO]);
```

> should this just set logger.level = Level.OFF when set enabled = false?

This works too.
Unfortunately, doing so would force the user to set `hierarchicalLoggingEnabled` to `true`. If not, `logging` throws:

```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following UnsupportedError was thrown building MyApp(dirty):
Unsupported operation: Please set "hierarchicalLoggingEnabled" to true if you want to change the
level on a non-root logger.
```

I'm not sure that acceptable to for the user to do that?

Reply:
I see, yeah this is fine